### PR TITLE
SLT: Set MZ_SYSTEM_PARAMETER_DEFAULT

### PR DESCRIPTION
**Pull request by def-**
As noticed by Dan

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by danhhz — APPROVED** [review]

**Comment by def-** [comment]
I find it suspicious that there were no errors, maybe the values are not propagated all the way yet.

**Comment by danhhz** [comment]
> stats_collection_enabled: Some(false), stats_filter_enabled: Some(false)

yeah, looks like it didn't take. I don't see any of these with true in the logs

File: src/sqllogictest/src/bin/sqllogictest.rs [hunk 1]
```diff
@@ -136,14 +136,26 @@ struct Args {
...
 }
 
 #[tokio::main]
 async fn main() -> ExitCode {
     mz_ore::panic::set_abort_on_panic();
     mz_ore::test::init_logging_default("warn");
 
-    let args: Args = cli::parse_args(CliConfig::default());
+    let args: Args = cli::parse_args(CliConfig {
+        env_prefix: Some("MZ_"),
```
Comment (def-) [inline src/sqllogictest/src/bin/sqllogictest.rs:156]: This part was missing, args are being passed through now.

**Comment by def-** [comment]
This slows down the fast SLT so much that is would have to be split into two parts. I can do that, but will wait if rest of QA team is fine with this change first.

File: ci/test/slt-fast.sh [hunk 2]
```diff
@@ -367,5 +367,6 @@ for f in "${tests[@]}"; do
 done
 tests=("${temp[@]}")
 
-sqllogictest -v --auto-index-selects "${tests[@]}" "$@"
-sqllogictest -v "${tests_without_views[@]}" "$@"
+#sqllogictest -v --auto-index-selects "${tests[@]}" "$@"
```
Comment (philip-stoev) [inline ci/test/slt-fast.sh:370]: Scalpel left in patient?

File: test/sqllogictest/cluster.slt [hunk 3]
```diff
@@ -34,9 +34,19 @@ ALTER CLUSTER foo RENAME TO bar
...
+simple conn=mz_system,user=mz_system
+ALTER SYSTEM SET enable_rbac_checks TO false;
+----
+COMPLETE 0
+
 statement error db error: ERROR: system cluster 'mz_introspection' cannot be modified
 ALTER CLUSTER mz_introspection RENAME TO foo
 
+simple conn=mz_system,user=mz_system
+ALTER SYSTEM SET enable_rbac_checks TO true;
```
Comment (philip-stoev) [inline test/sqllogictest/cluster.slt:46]: `ALTER SYSTEM RESET` may be safer generally.

**Review by philip-stoev — APPROVED** [review]

**Review by nrainer-materialize — APPROVED** [review]

File: ci/test/pipeline.template.yml [hunk 4]
```diff
@@ -277,9 +277,10 @@ steps:
       queue: linux-x86_64
 
   - id: sqllogictest-fast
-    label: Fast SQL logic tests
+    label: Fast SQL logic tests %n
     depends_on: build-x86_64
     timeout_in_minutes: 30
+    parallelism: 3
```
Comment (def-) [inline ci/test/pipeline.template.yml:283]: @philip-stoev Is this ok? With `parallelism: 2` one side got unlucky and still took 19 min.

**Comment by def-** [comment]
> This slows down the fast SLT so much that is would have to be split into two parts. I can do that, but will wait if rest of QA team is fine with this change first.

Now implemented in the second commit: https://github.com/MaterializeInc/materialize/pull/20451/commits/02d22e9e412d6fc44a151d54700fa340e13173e7

File: ci/test/pipeline.template.yml (hunk 4 above)
Comment (philip-stoev) [reply ci/test/pipeline.template.yml:283]: Yes more paralellism is fine with me. Ec2 now has per minute billing so the overhead from having more machines involved should be acceptable

File: test/sqllogictest/transform/predicate_pushdown.slt [hunk 5]
```diff
@@ -468,9 +468,19 @@ EOF
...
+simple conn=mz_system,user=mz_system
+ALTER SYSTEM SET enable_rbac_checks TO false;
+----
+COMPLETE 0
+
 statement ok
 DROP SCHEMA public CASCADE ;
 
+simple conn=mz_system,user=mz_system
+ALTER SYSTEM RESET enable_rbac_checks;
```
Comment (jkosh44) [inline test/sqllogictest/transform/predicate_pushdown.slt:480]: Ditto about the reset.

**Review by jkosh44 — APPROVED** [review]

File: test/sqllogictest/unstable.slt [hunk 6]
```diff
@@ -14,9 +14,19 @@ mode cockroach
...
+COMPLETE 0
+
 statement error cannot create index with unstable dependencies
 CREATE DEFAULT INDEX ON mz_internal.mz_view_foreign_keys
 
+simple conn=mz_system,user=mz_system
+ALTER SYSTEM RESET enable_rbac_checks;
+----
+COMPLETE 0
+
```
[the remark this answers is not in the document]
Comment (def-) [reply test/sqllogictest/unstable.slt:29]: In this change I have changed the default:
```
$ cat x.slt
query T
show enable_rbac_checks;
----
on

simple conn=mz_system,user=mz_system
ALTER SYSTEM SET enable_rbac_checks TO false;
----
COMPLETE 0

query T
show enable_rbac_checks;
----
off

simple conn=mz_system,user=mz_system
ALTER SYSTEM RESET enable_rbac_checks;
----
COMPLETE 0

query T
show enable_rbac_checks;
----
on

$ MZ_SYSTEM_PARAMETER_DEFAULT="enable_rbac_checks=true;enable_ld_rbac_checks=true" bin/sqllogictest -- x.slt
PASS: success=5 total=5
```

[the remark this answers is not in the document]
Comment (jkosh44) [reply test/sqllogictest/unstable.slt:29]: Ahh, I never knew we had the ability to alter the defaults for these.

**Pull request merged**